### PR TITLE
update dependencies, babel preset-env set to IE10

### PR DESCRIPTION
**Pull request by caub**
cf https://github.com/Financial-Times/polyfill-library/pull/59

I'm not sure this commit makes a real difference, the dist/ù files are exactly the same size with or without the "ie": "10" option in .babelrc, but let's try to polyfills.io

### CODE REVIEW

**Comment by caub** [comment]
I think we could remove and gitignore `package-lock.json`, it's not very useful for libraries like here

File: .babelrc [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {
   "presets": [
     [
-      "babel-preset-env",
+      "@babel/preset-env",
       {
-        "modules": false
+        "modules": false,
+        "targets": { "ie": "10" }
```
Comment (JakeChampion) [inline .babelrc]: https://polyfill.io supports IE/8 which is an ES3 environment whereas IE/10 is an ES5 environment.
The main difference between those environments for a compiler like BabelJS is in ES5, keywords can be used as property-names via dot-notation whereas in ES3 they have to use bracket-notation.

E.G. ES5 can have this code and ES3 can not have this code: `var obj = {}; obj.delete = 1;`

Comment (caub) [reply .babelrc]: well spotted https://github.com/babel/babel/issues/6606 suggests to add babel-plugin-transform-es3-member-expression-literals 

also do you want "targets": { "ie": "8" } (apparently the plugin above is still needed even with this)

**Comment by caub** [comment]
I updated with `"ie": "8"` as target and `"plugins": ["@babel/plugin-transform-member-expression-literals"]` but when running the build, the dist/ files are not modified, so nothing changed

**Comment by mo** [comment]
I tried to run abortcontroller-polyfill on IE8 and found that the function document.createEvent() doesn't exist, so I added a new path that calls into document.createEventObject() instead when document.createEvent() is missing.

After I did this, abortcontroller-polyfill worked on IE8 when I used it together with the "fetch-ie8" npm package. Here is the example:
https://github.com/mo/abortcontroller-polyfill-examples/tree/master/plain-javascript-fetch-ie8

github-fetch only supports IE 10+ though and while "unfetch" claims IE8 support in their README I could not get it to work using the latest version.

Also I could not get CORS requests to work on "fetch-ie8", and there is an open issue about unfetch with CORS on IE 8.

**Comment by mo** [comment]
@caub I've cleaned up the abortcontroller-polyfill test suite a bit now on master and upgraded it to webdriverio 5.x etc..

**Comment by mo** [comment]
I've merged this PR now and released it as 1.2.8

**Pull request closed without merging**